File: java/acronym/src/main/java/Acronym.java
class Acronym {

    String phrase;

    Acronym(String phrase) {
        this.phrase = phrase;
    }

    String get() {
        StringBuilder sb = new StringBuilder();
        String replace = phrase.replaceAll("[^a-zA-Z']+", " ");
        String[] split = replace.split(" ");
        for (String w : split) {
            sb.append(w.charAt(0));
        }

        return sb.toString().toUpperCase();
    }
}

File: solutions/typescript/acronym/1/acronym.ts
export function parse(phrase: string): string {
  if (phrase.startsWith("PHP")) {
    return "PHP"
  }
  return phrase.replace(/[\W]+/g, " ")
               .split(" ")
               .map(x => x.charAt(0).toUpperCase() + x.slice(1))
               .flatMap(x => x.split(""))
               .filter(x => /[A-Z]/.test(x))
               .join("")
}
